### ISSUE TITLE
Support "Want AuthnRequests Signed" for SAML Identity Provider resource

### ISSUE THREAD

**Issue by tpauling** [mrparkers/terraform-provider-keycloak#279]
Seems the setting of "Want AuthnRequests Signed" is missing for a SAML indentity provider. Adding `want_authn_requests_signed` to the resource seems not working with latest release v1.18.0.

```
on saml.tf line 13, in resource "keycloak_saml_identity_provider" "realm_identity_provider":
  13:   want_authn_requests_signed = false
An argument named "want_authn_requests_signed" is not expected here.
```

**Comment by tomrutsaert** [comment]
I took a look in the code, and I found following snippet:

```
	if _, ok := data.GetOk("signature_algorithm"); ok {
		rec.Config.WantAuthnRequestsSigned = true
	}
```

It seems the code will set it to true when there is a signature algorithm in the config.

**Comment by tpauling** [comment]
Sometimes you need an expert to find this kind of things. Was not expecting this logic. After removing the `signature_algorithm` from the resource it worked out. Thanks for helping!

**Comment by tomrutsaert** [comment]
Imho there should not be such logic in the provider as you never know how keycloak interpretation might change.
At least it should be added to the documentation

**Comment by tpauling** [comment]
yep agree. Since the logic of the UI in keycloak behaves differently. only if you enable `want_authn_requests_signed` there other fields are appearing (incl. signature_algorithm)

**Comment by davide-talesco** [comment]
can then that conditional assignment be removed?
